lib/elements/reinforcement/straight_longitudinal.rb

```diff
@@ -0,0 +1,98 @@
+require 'elements/reinforcement/base'
+require 'elements/reinforcement/straight_longitudinal_layer'
+require 'errors/reinforcement/empty_layers'
+
+module Elements
+  module Reinforcement
+    class StraightLongitudinal < Base
+      def initialize(distribution_direction:, above_middle: false)
+        @above_middle = above_middle
+        @layers = []
+        @distribution_direction = distribution_direction
+      end
+
+      def add_layer(start_location:, end_location:, amount_of_rebars:, rebar:)
+        id = @layers.empty? ? 1 : @layers.last.id
+        start_location = modify_value_3_of_location(start_location, 0.5 * rebar.diameter)
+        end_location = modify_value_3_of_location(end_location, 0.5 * rebar.diameter)
+        @layers << Elements::Reinforcement::StraightLongitudinalLayer.new(
+          start_location:,
+          end_location:,
+          amount_of_rebars:,
+          rebar:,
+          id:,
+          distribution_direction: @distribution_direction
+        )
+
+        @layers.last
+      end
+
+      def change_layer_rebar_configuration(
+        id_of_layer_to_change:,
+        amount_of_new_rebars:,
+        new_rebar:
+      )
+        straight_longitudinal_layer = find(id_of_layer_to_change)
+
+        offset = (straight_longitudinal_layer.diameter - new_rebar.diameter) / 2
+        offset *= -1 unless @above_middle
+
+        straight_longitudinal_layer.modify_rebar_configuration(
+          amount_of_new_rebars:,
+          new_rebar:,
+          offset:
+        )
+
+        straight_longitudinal_layer
+      end
```

Comment:
@mapra99 

can you take a look to [--> our wiki <-- Elements::Reinforcement::StraightLongitudinal #change_layer_rebar_configuration](https://pradaing.notion.site/Reinforcement-ff39ec7738fe4692a807ec9eaa1063ab#6def4b45b4c74633a0c7fcd9dd02989c)